tests/autoscaling/test_autoscaling_service_lib.py

```diff
@@ -315,13 +329,51 @@ def test_http_metrics_provider_no_data():
         branch_dict={},
     )
     fake_marathon_tasks = [mock.Mock(id='fake-service.fake-instance', host='fake_host', ports=[30101])]
-    mock_request_result = mock.Mock(json=mock.Mock(return_value='malformed_result'))
-    with mock.patch('paasta_tools.autoscaling.autoscaling_service_lib.requests.get',
-                    autospec=True, return_value=mock_request_result):
+    mock_json_mapper = mock.Mock(side_effect=autoscaling_service_lib.MetricsProviderNoDataError)
+
+    with mock.patch('paasta_tools.autoscaling.autoscaling_service_lib.get_json_body_from_service', autospec=True):
         with raises(autoscaling_service_lib.MetricsProviderNoDataError):
-            autoscaling_service_lib.http_metrics_provider(fake_marathon_service_config,
-                                                          fake_marathon_tasks,
-                                                          mock.Mock()) == 0.5
+            autoscaling_service_lib.get_http_utilization_for_all_tasks(
+                fake_marathon_service_config,
+                fake_marathon_tasks,
+                endpoint='fake-endpoint',
+                json_mapper=mock_json_mapper,
+            )
+
+
+def test_http_metrics_provider():
+    fake_marathon_tasks = [mock.Mock(id='fake-service.fake-instance', host='fake_host', ports=[30101])]
+
+    with mock.patch(
+        'paasta_tools.autoscaling.autoscaling_service_lib.get_json_body_from_service',
+        autospec=True,
+    ) as mock_get_json_body_from_service:
+        mock_get_json_body_from_service.return_value = {'utilization': 0.5}
+        assert autoscaling_service_lib.http_metrics_provider(
+            marathon_service_config=mock.Mock(),
+            marathon_tasks=fake_marathon_tasks,
+        ) == 0.5
+
+
+def test_uwsgi_metrics_provider():
+    fake_marathon_tasks = [mock.Mock(id='fake-service.fake-instance', host='fake_host', ports=[30101])]
+
+    with mock.patch(
+        'paasta_tools.autoscaling.autoscaling_service_lib.get_json_body_from_service',
+        autospec=True,
+    ) as mock_get_json_body_from_service:
+        mock_get_json_body_from_service.return_value = {
+            'workers': [
+                {'status': 'idle'},
+                {'status': 'not-idle'},
+                {'status': 'not-idle'},
+                {'status': 'not-idle'},
```

Comment:
I _believe_ the only two states are `busy` and `idle`. however in previous code, I have just used `busy` and `!= busy`. Either way, `not-idle` is not a valid state